pkg/controller/logcollector/logcollector_controller.go

```diff
@@ -542,21 +578,58 @@ func (r *ReconcileLogCollector) Reconcile(ctx context.Context, request reconcile
 	}
 
 	var eksConfig *render.EksCloudwatchLogConfig
-	if installation.KubernetesProvider == operatorv1.ProviderEKS {
-		log.Info("Managed kubernetes EKS found, getting necessary credentials and config")
-		if instance.Spec.AdditionalSources != nil {
-			if instance.Spec.AdditionalSources.EksCloudwatchLog != nil {
-				eksConfig, err = getEksCloudwatchLogConfig(r.client,
-					instance.Spec.AdditionalSources.EksCloudwatchLog.FetchInterval,
-					instance.Spec.AdditionalSources.EksCloudwatchLog.Region,
-					instance.Spec.AdditionalSources.EksCloudwatchLog.GroupName,
-					instance.Spec.AdditionalSources.EksCloudwatchLog.StreamPrefix)
-				if err != nil {
-					r.status.SetDegraded(operatorv1.ResourceReadError, "Error retrieving EKS Cloudwatch Logs configuration", err, reqLogger)
-					return reconcile.Result{}, err
+	if variant == operatorv1.TigeraSecureEnterprise {
+		if installation.KubernetesProvider == operatorv1.ProviderEKS {
+			log.Info("Managed kubernetes EKS found, getting necessary credentials and config")
+			if instance.Spec.AdditionalSources != nil {
+				if instance.Spec.AdditionalSources.EksCloudwatchLog != nil {
+					eksConfig, err = getEksCloudwatchLogConfig(r.client,
+						instance.Spec.AdditionalSources.EksCloudwatchLog.FetchInterval,
+						instance.Spec.AdditionalSources.EksCloudwatchLog.Region,
+						instance.Spec.AdditionalSources.EksCloudwatchLog.GroupName,
+						instance.Spec.AdditionalSources.EksCloudwatchLog.StreamPrefix)
+					if err != nil {
+						r.status.SetDegraded(operatorv1.ResourceReadError, "Error retrieving EKS Cloudwatch Logs configuration", err, reqLogger)
+						return reconcile.Result{}, err
+					}
 				}
 			}
 		}
+	} else {
+		// patch and get the felix configuration
```

Comment:
All these configs are enabled by default for Enterprise.

Reply:
Where does that defaulting happen? 

Sounds like there isn't a problem with us executing this code for both OSS and Enterprise? 

The more we let the OSS / enterprise implementation diverge, the harder it comes to maintain and understand those differences. So we should be doing whatever we can to avoid these sorts of different codepaths and implementations.

It sounds like we should be putting this in the path for both OSS and Enterprise and making the implementations consistent.

Reply:
It shouldn't be a problem executing this code for Enterprise and OSS.